homeassistant/core.py

```diff
@@ -1257,7 +1257,7 @@ def __init__(
                 "State max length is 255 characters."
             )
 
-        self.entity_id = entity_id.lower()
+        self.entity_id = entity_id
```

Comment:
The rest of this PR makes sense to me, but why is ".lower()" no longer needed ? it might be correct but looks suspicious

Reply:
Its already lowered here https://github.com/home-assistant/core/blob/b752419f25650f8a0f3aded6b2e883f154a95922/homeassistant/core.py#L1610

Reply:
nice catch!